src/WebSockets.jl

```diff
@@ -94,22 +96,27 @@ function write_fragment(io::IO, islast::Bool, data::Array{UInt8}, opcode)
   # 2. set a mask
   # 3. xor data with mask
 
-  if l <= 125
-    write(io, b1)
-    write(io, @compat UInt8(l))
-    write(io, data)
-  elseif l <= typemax(UInt16)
-    write(io, b1)
-    write(io, @compat UInt8(126))
-    write(io, hton(@compat UInt16(l)))
-    write(io, data)
-  elseif l <= typemax(UInt64)
-    write(io, b1)
-    write(io, @compat UInt8(127))
-    write(io, hton(@compat UInt64(l)))
-    write(io, data)
-  else
-    error("Attempted to send too much data for one websocket fragment\n")
+  isa(io, TCPSock) && lock(io.lock)
+  try
+    if l <= 125
+      write(io, b1)
+      write(io, @compat UInt8(l))
+      write(io, data)
+    elseif l <= typemax(UInt16)
+      write(io, b1)
+      write(io, @compat UInt8(126))
+      write(io, hton(@compat UInt16(l)))
+      write(io, data)
+    elseif l <= typemax(UInt64)
+      write(io, b1)
+      write(io, @compat UInt8(127))
+      write(io, hton(@compat UInt64(l)))
+      write(io, data)
+    else
+      error("Attempted to send too much data for one websocket fragment\n")
+    end
+  finally
```

Comment:
No, but then 0.3 does not have `lock` either. For 0.3, you can do multiple writes to a IOBuffer, followed by a takebuf_array and write the array once to the socket. The IOBuffer writes will not yield to other tasks, so you will not need a lock.